source/service.go

```diff
@@ -75,6 +78,10 @@ func (sc *serviceSource) Endpoints() ([]*endpoint.Endpoint, error) {
 	if err != nil {
 		return nil, err
 	}
+	services.Items, err = sc.filterByAnnotations(services.Items, sc.annotationFilter)
```

Comment:
here is the same case, passing sc.annotationFilter to a function receiver of sc.